pulsar/producer_partition.go

```diff
@@ -934,3 +941,17 @@ func (i *pendingItem) Complete() {
 	i.completed = true
 	buffersPool.Put(i.batchData)
 }
+
+func (p *partitionProducer) closFailedProducer() {
+	p.setProducerState(producerFailed)
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	cp := &closeProducer{&wg}
+	p.eventsChan <- cp
+
+	go p.runEventsLoop()
```

Comment:
Should we prefer to call `internalClose(&wg)` directly here, instead of using the channel and running the events loop?

My instinct is that `runEventsLoop` would be reserved for fully initialized producers.

Reply:
You are correct - this is really an exception so we can go direct to the internal close.